### PR TITLE
Redis replicationFactor/replicaNodes immutable

### LINKED ISSUE #676: Redis replicationFactor/replicaNodes immutable
The replicationFactor and replicaNodes field of the redis DC immutable.

### PR DESCRIPTION
**Pull request by worryg0d**
This PR provides immutability for the `replicationFactor` and `replicaNodes` fields of the Redis data centre.

Also, it provides automatical reconcile of external changes of the k8s resource spec equals to instaclustr resource spec.

closes #676

### CODE REVIEW

File: apis/clusters/v1beta1/redis_webhook.go [hunk 1]
```diff
@@ -63,10 +64,13 @@ func (r *Redis) Default() {
...
-	}
-
-	redislog.Info("default values are set", "name", r.Name)
 
+		if dataCentre.ReplicationFactor > 0 && dataCentre.NodesNumber == 0 {
+			dataCentre.NodesNumber = dataCentre.MasterNodes * dataCentre.ReplicationFactor
+		} else if dataCentre.NodesNumber > 0 && dataCentre.ReplicationFactor == 0 {
+			dataCentre.ReplicationFactor = dataCentre.NodesNumber / dataCentre.MasterNodes
+		}
+	}
```
Comment (testisnullus) [inline apis/clusters/v1beta1/redis_webhook.go:73]: If the replication factor and nodes number are equal to 0, I assume that we create master nodes only, am I right?

Comment (worryg0d) [reply apis/clusters/v1beta1/redis_webhook.go:73]: Yes, you are.

**Review by tengu-alt — APPROVED** [review]

File: controllers/clusters/redis_controller.go [hunk 2]
```diff
@@ -869,9 +869,22 @@ func (r *RedisReconciler) newWatchStatusJob(redis *v1beta1.Redis) scheduler.Job
 			}
 		}
 
-		if iRedis.Status.CurrentClusterOperationStatus == models.NoOperation &&
+		equals := redis.Spec.IsEqual(iRedis.Spec)
```
Comment (ribaraka) [inline controllers/clusters/redis_controller.go:872]: please add this logic to the rest `Kind`s

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/redis_webhook.go [hunk 3]
```diff
@@ -64,6 +64,17 @@ func (r *Redis) Default() {
 
 	for _, dataCentre := range r.Spec.DataCentres {
 		dataCentre.SetDefaultValues()
+
+		fmt.Println("DEFAULTER")
+		fmt.Println("nodes number", dataCentre.NodesNumber)
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply apis/clusters/v1beta1/redis_webhook.go]: fixed

File: controllers/clusters/redis_controller.go (hunk 2 above)
Comment (worryg0d) [reply controllers/clusters/redis_controller.go]: already fixed

**Review by ribaraka — APPROVED** [review]

**Pull request merged**